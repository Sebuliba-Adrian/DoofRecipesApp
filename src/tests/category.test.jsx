import React from "react";
import renderer from "react-test-renderer";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Category from "../components/categories/category";

const props = {
  key: 1,
  category: {
    id: 1,
    name: "Test category",
    description: "This is a test category",
    date_created: "Wed, 20 Feb 2018 11:31:50 GMT",
    date_modified: "Wed, 20 Feb 2018 11:31:50 GMT"
  },
  request: jest.fn(),
  viewRecipes: jest.fn()
};

describe("Test recipes", () => {
  it("renders correctly", () => {
    const category = renderer.create(<category {...props} />).toJSON();
    expect(category).toMatchSnapshot();
  });

  configure({ adapter: new Adapter() });
  const category = mount(<Category {...props} />);
  global.fetch = jest.fn(() => {
    return Promise.resolve({
      json: () => Promise.resolve({})
    });
  });
  it("Shows  recipes", () => {
    const categoryrow = category.find("a");
    categoryrow.simulate("click");
    expect(props.viewRecipes).toHaveBeenCalled();
  });
});
