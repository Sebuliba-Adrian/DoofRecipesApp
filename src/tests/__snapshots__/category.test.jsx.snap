// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test recipes renders correctly 1`] = `
<category
  category={
    Object {
      "date_created": "Wed, 20 Feb 2018 11:31:50 GMT",
      "date_modified": "Wed, 20 Feb 2018 11:31:50 GMT",
      "description": "This is a test category",
      "id": 1,
      "name": "Test category",
    }
  }
  request={[Function]}
  viewRecipes={[Function]}
/>
`;
